packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -224,13 +236,18 @@ class PointerEventResampler {
       // Skip `move` and `hover` events as they are automatically
       // generated when the position has changed.
       if (event is! PointerMoveEvent && event is! PointerHoverEvent) {
+        // Add synthetics `move` or `hover` event if position has changed.
+        if (position != _position) {
+          final Offset delta = position - _position;
+          callback(_toMoveOrHoverEvent(event, position, delta, _pointerIdentifier, sampleTime, wasDown));
```

Comment:
Like here, we need a comment saying that for device without hover event, add event and down event will always have the same position and therefore won't add hover event

Reply:
Done